packages/synapse-react-client/src/components/TableColumnSchemaEditor/TableColumnSchemaEditorUtils.ts

```diff
@@ -263,6 +263,19 @@ export function canHaveRestrictedValues(
   }
 }
 
+function transformEnumValues(
+  enumValues: string[],
+  columnType: ColumnType | ColumnTypeEnum,
+): (string | number)[] {
+  // SWC-6622 - Special case: if these are INTEGER, convert to numbers before inserting data into form components
+  // The editor validates that these are integers, so inserting the strings returned by the API causes an
+  // unexpected validation error
+  if (columnType === ColumnTypeEnum.INTEGER) {
+    return enumValues.map(value => parseInt(value))
+  }
+  return enumValues
```

Comment:
The only column types that can have enumValues are STRING, INTEGER, ENTITYID, so this is the only special case